### ISSUE TITLE
[bug] Provenance generation fails after updating the generator to v1.9.0

### ISSUE THREAD

**Issue by behnazh-w** [slsa-framework/slsa-github-generator#2691]
The provenance generation on [micronaut-core](https://github.com/micronaut-projects/micronaut-core/actions/runs/6039112753/job/16391009301) repository has failed after [updating](https://github.com/micronaut-projects/micronaut-core/actions/runs/6039112753/workflow#L137) the provenance generator to v1.9.0 and the error message is not helpful for debugging.

**Comment by laurentsimon** [comment]
Thanks for the issue @behnazh-w . Error is `GitHub Actions is temporarily unavailable` which is not under our control, afaik.

@steiza is there anything we can do? Is it something the repository uses can do (micronaut-core)?

**Comment by steiza** [comment]
I'm missing some context here. I think this is the error in question?

```
verify: creating verifier: 503 Service Unavailable: { "message": "GitHub Actions is temporarily unavailable. Please visit https://www.githubstatus.com/ for the status of our services.", "ref": "Ref A: AFAEB017732F4791BCCA752C39D20642 Ref B: BY3EDGE0120 Ref C: 2023-08-31T17:07:15Z" }
```

What is `verify`? Is that a CLI tool that's calling GitHub APIs?

There will definitely be intermittent (and hopefully short!) cases where GitHub APIs are not available. Of course it depends on the tolerance your application has, but I think it's reasonable to retry a few times over a few minutes before giving up and reporting a failure to the end user.

**Comment by laurentsimon** [comment]
You're correct, the code is https://github.com/slsa-framework/slsa-github-generator/blob/main/github/oidc.go#L173-L178. Thanks for the tip, cc'ing @ianlewis for the retry
